Import element tests from TestBench

Artur- opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/framework/8117)
<!-- Reviewable:end -->

Artur- committed:
Import element tests from TestBench

Artur- committed:
More tests

hesara commented:
Based on partial review: These seem incomplete (e.g. Upload and some other components missing) but are an improvement over the current situation. Copyright years should be updated in all files, and ideally broken class level javadoc filled in or removed. Javadoc since tags can be filled in with 8.0 if relevant, removed otherwise.

---

Reviewed 26 of 109 files at r1.
Review status: 26 of 109 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8117#-:-K_TKbEe18FbM0vc2GMj:b-4dztdc)*
<!-- Sent from Reviewable.io -->

hesara commented:
I'll comment here on various issues found and then perhaps make a follow-up patch in this PR fixing some of the issues to get this merged. #8112 will then need to be adjusted to fix a few of the issues here.

---

Reviewed 24 of 109 files at r1.
Review status: 50 of 109 files reviewed at latest revision, 13 unresolved discussions.

---

*[uitest/src/main/java/com/vaadin/tests/elements/calendar/CalendarUI.java, line 22 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TM49KjqBrcBNYDDtI:-K_TM49KjqBrcBNYDDtJ:b3f8gu3) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/main/java/com/vaadin/tests/elements/calendar/CalendarUI.java#L22)):*
> ```Java
>             @Override
>             public void buttonClick(Button.ClickEvent event) {
>                 java.util.Calendar javaCalendar = java.util.Calendar
> ```

not deterministic, locale hasn't been set

---

*[uitest/src/main/java/com/vaadin/tests/elements/checkboxgroup/CheckBoxGroupSetValue.java, line 25 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TMzlJYpnnZbPXQQeb:-K_TMzlK5wL0ZcUX-som:bna2vt2) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/main/java/com/vaadin/tests/elements/checkboxgroup/CheckBoxGroupSetValue.java#L25)):*
> ```Java
>     @Override
>     protected String getTestDescription() {
>         return "Test CheckBoxGroup element setValue() and SelectByText()";
> ```

selectByText()

---

*[uitest/src/main/java/com/vaadin/tests/elements/datefield/DateFieldPopupSetValue.java, line 16 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TSoY9eWnUeyGflubK:-K_TSoY9eWnUeyGflubL:b2ic3p1) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/main/java/com/vaadin/tests/elements/datefield/DateFieldPopupSetValue.java#L16)):*
> ```Java
> public class DateFieldPopupSetValue extends AbstractTestUI {
> 
>     public Calendar calendar = Calendar.getInstance();
> ```

unused field

---

*[uitest/src/main/java/com/vaadin/tests/elements/grid/GridUI.java, line 58 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TUX0eQEMcDceVZvLP:-K_TUX0eQEMcDceVZvLQ:bb1h1ma) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/main/java/com/vaadin/tests/elements/grid/GridUI.java#L58)):*
> ```Java
>     }
> 
>     private class Item {
> ```

(probably should be static nested class)

---

*[uitest/src/main/java/com/vaadin/tests/elements/listselect/ListSelectOptionClick.java, line 25 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TXL3jhdqS5WG7gLs1:-K_TXL3jhdqS5WG7gLs2:b-bwz707) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/main/java/com/vaadin/tests/elements/listselect/ListSelectOptionClick.java#L25)):*
> ```Java
>         options.add("item2");
>         options.add("item3");
>         multiSelect.setDataProvider(new ListDataProvider<String>(options));
> ```

(could also simply use multiSelect.setItems(options))

---

*[uitest/src/main/java/com/vaadin/tests/elements/listselect/ListSelectOptionClick.java, line 38 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TXWh9FaURytWEzknY:-K_TXWh9FaURytWEzknZ:bqrari8) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/main/java/com/vaadin/tests/elements/listselect/ListSelectOptionClick.java#L38)):*
> ```Java
>     @Override
>     protected String getTestDescription() {
>         return "Test that user can pick option from ListSelectElement by call click() method";
> ```

by calling

---

*[uitest/src/test/java/com/vaadin/tests/elements/checkboxgroup/CheckBoxGroupSetValueTest.java, line 26 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TNKgHbWUyMOsP7leq:-K_TNKgHbWUyMOsP7ler:b-3kw20v) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/test/java/com/vaadin/tests/elements/checkboxgroup/CheckBoxGroupSetValueTest.java#L26)):*
> ```Java
>     public void testSetValue() {
>         group.setValue(NEW_VALUE);
>         Assert.assertEquals(NEW_VALUE, group.getValue());
> ```

getValue() assumes single value selection? The framework project custom element for CBG does have getSelection() that returns a List, unified with the TB element in #8112 . Same also below. (probably to be addressed after this is merged)

---

*[uitest/src/test/java/com/vaadin/tests/elements/combobox/ComboBoxInputNotAllowedTest.java, line 16 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TQufFKTN_AimCLrjN:-K_TQufFKTN_AimCLrjO:biz1ghq) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/test/java/com/vaadin/tests/elements/combobox/ComboBoxInputNotAllowedTest.java#L16)):*
> ```Java
>         openTestURL();
>         ComboBoxElement cb = $(ComboBoxElement.class).first();
>         cb.selectByText("Foobar");
> ```

(not really testing anything except that no exception)

---

*[uitest/src/test/java/com/vaadin/tests/elements/combobox/ComboBoxUITest.java, line 54 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TRbOESMWiKpOYjHWQ:-K_TRbOESMWiKpOYjHWR:bvkcnce) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/test/java/com/vaadin/tests/elements/combobox/ComboBoxUITest.java#L54)):*
> ```Java
>     public void testSelectByTextNotFound() {
>         ComboBoxElement cb = $(ComboBoxElement.class).first();
>         cb.selectByText("foobar");
> ```

(not checking the selection, just that there are no exceptions)

---

*[uitest/src/test/java/com/vaadin/tests/elements/grid/GridUITest.java, line 14 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TWvvkIq4Lh-CLbsMk:-K_TWvvkIq4Lh-CLbsMl:bu6a0dy) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/test/java/com/vaadin/tests/elements/grid/GridUITest.java#L14)):*
> ```Java
> 
>     @Test
>     @Ignore("API for getRowCount is missing in FW8 ATM")
> ```

to address - perhaps with/after #8112

---

*[uitest/src/test/java/com/vaadin/tests/elements/grid/GridUITest.java, line 35 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TX1QSXlLhhJsfkdPG:-K_TX1QSXlLhhJsfkdPH:bu6a0dy) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/test/java/com/vaadin/tests/elements/grid/GridUITest.java#L35)):*
> ```Java
> 
>     @Test
>     @Ignore("API needed for getRows working is missing in FW8 ATM")
> ```

to address - perhaps with/after #8112

---

*[uitest/src/test/java/com/vaadin/tests/elements/listselect/ListSelectOptionClickTest.java, line 28 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TZDA71d-CT9hLY1rY:-K_TZDA71d-CT9hLY1rZ:b3f2ouc) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/test/java/com/vaadin/tests/elements/listselect/ListSelectOptionClickTest.java#L28)):*
> ```Java
> 
>     @Test
>     @Ignore("depends on framework8-issues/issues/464 fix")
> ```

(to add a comment in vaadin/framework8-issues#464 about re-enabling these tests)

---

*[uitest/src/test/java/com/vaadin/tests/elements/listselect/ListSelectOptionClickTest.java, line 77 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TZSpKoZzrrakJg3oQ:-K_TZSpKoZzrrakJg3oR:b-ng3jbo) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/test/java/com/vaadin/tests/elements/listselect/ListSelectOptionClickTest.java#L77)):*
> ```Java
>                 "The number of list select valueChange events is not one.",
>                 "1: item2", actualCounter);
>     }
> ```

(no test for selecting/deselecting the same item twice, not sure if that could be an issue)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8117#-:-K_TTzD7-JbvcCzlboOG:burv133)*
<!-- Sent from Reviewable.io -->

Artur- commented:
Review status: 50 of 109 files reviewed at latest revision, 12 unresolved discussions.

---

*[uitest/src/main/java/com/vaadin/tests/elements/calendar/CalendarUI.java, line 22 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TM49KjqBrcBNYDDtI:-K_Tm6i2aXEsaYTb7rWy:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/main/java/com/vaadin/tests/elements/calendar/CalendarUI.java#L22)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

not deterministic, locale hasn't been set
</blockquote></details>

Done.

---

*[uitest/src/main/java/com/vaadin/tests/elements/checkboxgroup/CheckBoxGroupSetValue.java, line 25 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TMzlJYpnnZbPXQQeb:-K_Tq1GInCvL1WnO3oVo:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/main/java/com/vaadin/tests/elements/checkboxgroup/CheckBoxGroupSetValue.java#L25)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

selectByText()
</blockquote></details>

Done.

---

*[uitest/src/main/java/com/vaadin/tests/elements/datefield/DateFieldPopupSetValue.java, line 16 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TSoY9eWnUeyGflubK:-K_TmC943LpNAQIMl2NY:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/main/java/com/vaadin/tests/elements/datefield/DateFieldPopupSetValue.java#L16)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

unused field
</blockquote></details>

Done.

---

*[uitest/src/main/java/com/vaadin/tests/elements/grid/GridUI.java, line 58 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TUX0eQEMcDceVZvLP:-K_TmJJkddt8iK2lPsr1:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/main/java/com/vaadin/tests/elements/grid/GridUI.java#L58)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

(probably should be static nested class)
</blockquote></details>

Done.

---

*[uitest/src/main/java/com/vaadin/tests/elements/listselect/ListSelectOptionClick.java, line 25 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TXL3jhdqS5WG7gLs1:-K_TmQR7olEDo2218-Vk:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/main/java/com/vaadin/tests/elements/listselect/ListSelectOptionClick.java#L25)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

(could also simply use multiSelect.setItems(options))
</blockquote></details>

Done.

---

*[uitest/src/main/java/com/vaadin/tests/elements/listselect/ListSelectOptionClick.java, line 38 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TXWh9FaURytWEzknY:-K_TmTLzy8IXA3EIBNJE:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/main/java/com/vaadin/tests/elements/listselect/ListSelectOptionClick.java#L38)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

by calling
</blockquote></details>

Done.

---

*[uitest/src/test/java/com/vaadin/tests/elements/checkboxgroup/CheckBoxGroupSetValueTest.java, line 26 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TNKgHbWUyMOsP7leq:-K_Tmka6zEL3FlXMY0JF:b-a00o09) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/test/java/com/vaadin/tests/elements/checkboxgroup/CheckBoxGroupSetValueTest.java#L26)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

getValue() assumes single value selection? The framework project custom element for CBG does have getSelection() that returns a List, unified with the TB element in #8112 . Same also below. (probably to be addressed after this is merged)
</blockquote></details>

`getValue` makes no sense, imported `getSelection` from #8112 instead

---

*[uitest/src/test/java/com/vaadin/tests/elements/combobox/ComboBoxInputNotAllowedTest.java, line 16 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TQufFKTN_AimCLrjN:-K_Tom07UTIuVRtH2var:b-v9vfnn) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/test/java/com/vaadin/tests/elements/combobox/ComboBoxInputNotAllowedTest.java#L16)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

(not really testing anything except that no exception)
</blockquote></details>

This is waiting for https://github.com/vaadin/testbench/pull/835, which needs to be fixed for all components and changed into a FW PR now

---

*[uitest/src/test/java/com/vaadin/tests/elements/combobox/ComboBoxUITest.java, line 54 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TRbOESMWiKpOYjHWQ:-K_Tq2gJwCmN-evhJXcP:bnj7hl0) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/test/java/com/vaadin/tests/elements/combobox/ComboBoxUITest.java#L54)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

(not checking the selection, just that there are no exceptions)
</blockquote></details>

same as above

---

*[uitest/src/test/java/com/vaadin/tests/elements/grid/GridUITest.java, line 14 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TWvvkIq4Lh-CLbsMk:-K_TozdQ3NPdDuA5t-Du:b-wsbl8t) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/test/java/com/vaadin/tests/elements/grid/GridUITest.java#L14)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

to address - perhaps with/after #8112
</blockquote></details>

The API has already been merged at this point

---

*[uitest/src/test/java/com/vaadin/tests/elements/grid/GridUITest.java, line 35 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TX1QSXlLhhJsfkdPG:-K_Tp4nmqtXbjdOnbyAT:b-wsbl8t) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/test/java/com/vaadin/tests/elements/grid/GridUITest.java#L35)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

to address - perhaps with/after #8112
</blockquote></details>

The API has already been merged at this point

---

*[uitest/src/test/java/com/vaadin/tests/elements/listselect/ListSelectOptionClickTest.java, line 28 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TZDA71d-CT9hLY1rY:-K_TpUnAMn9A6JCmYZLh:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/test/java/com/vaadin/tests/elements/listselect/ListSelectOptionClickTest.java#L28)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

(to add a comment in vaadin/framework8-issues#464 about re-enabling these tests)
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8117)*
<!-- Sent from Reviewable.io -->

hesara commented:
Some more comments, more minor comments are likely as continuing review.

---

Reviewed 10 of 109 files at r1, 6 of 9 files at r2.
Review status: 59 of 110 files reviewed at latest revision, 2 unresolved discussions.

---

*[uitest/src/main/java/com/vaadin/tests/elements/slider/SliderGetHandle.java, line 24 at r2](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_Tuu_R9nj7Fyjch5h2:-K_Tuu_R9nj7Fyjch5h3:b-l9og7j) ([raw file](https://github.com/vaadin/framework/blob/82bc85227000f1a91f9fc6d825e310c3eb91ceae/uitest/src/main/java/com/vaadin/tests/elements/slider/SliderGetHandle.java#L24)):*
> ```Java
>     @Override
>     protected String getTestDescription() {
>         return "Check getHanler method works. Get handler and move it, check get Value return new value";
> ```

getHandler()?? handle?

---

*[uitest/src/test/java/com/vaadin/tests/elements/panel/PanelScrollTest.java, line 35 at r2](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TuYyg9M3c_EvhkW66:-K_TuYyg9M3c_EvhkW67:b-83zvon) ([raw file](https://github.com/vaadin/framework/blob/82bc85227000f1a91f9fc6d825e310c3eb91ceae/uitest/src/test/java/com/vaadin/tests/elements/panel/PanelScrollTest.java#L35)):*
> ```Java
>         JavascriptExecutor js = getCommandExecutor();
>         String jsScript = "return arguments[0].getElementsByClassName(\"v-scrollable\")[0].scrollTop;";
>         if (BrowserUtil.isIE8(getDesiredCapabilities())) {
> ```

IE8 not supported by FW8 here and below, so could also simplify this

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8117#-:-K_Tv4M9Fpc6lzhfXj6D:byym4pm)*
<!-- Sent from Reviewable.io -->

Artur- commented:
Review status: 58 of 110 files reviewed at latest revision, 2 unresolved discussions.

---

*[uitest/src/main/java/com/vaadin/tests/elements/slider/SliderGetHandle.java, line 24 at r2](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_Tuu_R9nj7Fyjch5h2:-K_U41RI1js46hMg9_u6:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/82bc85227000f1a91f9fc6d825e310c3eb91ceae/uitest/src/main/java/com/vaadin/tests/elements/slider/SliderGetHandle.java#L24)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

getHandler()?? handle?
</blockquote></details>

Done.

---

*[uitest/src/test/java/com/vaadin/tests/elements/panel/PanelScrollTest.java, line 35 at r2](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TuYyg9M3c_EvhkW66:-K_U1UzLyh322doXtkDA:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/82bc85227000f1a91f9fc6d825e310c3eb91ceae/uitest/src/test/java/com/vaadin/tests/elements/panel/PanelScrollTest.java#L35)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

IE8 not supported by FW8 here and below, so could also simplify this
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8117)*
<!-- Sent from Reviewable.io -->

hesara commented:
Reviewed 34 of 109 files at r1, 2 of 9 files at r2, 1 of 2 files at r3.
Review status: 95 of 110 files reviewed at latest revision, 12 unresolved discussions.

---

*[testbench-api/src/main/java/com/vaadin/testbench/elements/CheckBoxGroupElement.java, line 62 at r3](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_ciSzb3T1rU-nadnd-:-K_ciSzb3T1rU-nadnd0:bazorkc) ([raw file](https://github.com/vaadin/framework/blob/0740ac116e637d81bd7e2415746f020dd25102e6/testbench-api/src/main/java/com/vaadin/testbench/elements/CheckBoxGroupElement.java#L62)):*
> ```Java
>      * @return value of the selected option in the option group
>      */
>     public String getValue() {
> ```

(I hope no tests depend on this, probably not an issue)

---

*[uitest/src/main/java/com/vaadin/tests/elements/button/ButtonUI.java, line 34 at r3](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_cTNjI4CQ-BvafSrar:-K_cTNjI4CQ-BvafSras:b-nok2hb) ([raw file](https://github.com/vaadin/framework/blob/ad3a8851f58992099c897a905168db6829406f5a/uitest/src/main/java/com/vaadin/tests/elements/button/ButtonUI.java#L34)):*
> ```Java
>     public static final String TEXT_FIELD_ID = "testTextfield";
>     public static final String LABEL_ID = "testLabel";
>     public static String QUITE_BUTTON_ID = "quiteButton";
> ```

I guess "quite" means "quiet" all over this file, but not a blocker

---

*[uitest/src/test/java/com/vaadin/tests/elements/abstracttextfield/AbstractFieldElementSetValueReadOnlyTest.java, line 16 at r3](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_cRwVEFh9x6Mmteh4S:-K_cRwVEFh9x6Mmteh4T:b-tqcxz3) ([raw file](https://github.com/vaadin/framework/blob/ad3a8851f58992099c897a905168db6829406f5a/uitest/src/test/java/com/vaadin/tests/elements/abstracttextfield/AbstractFieldElementSetValueReadOnlyTest.java#L16)):*
> ```Java
> import com.vaadin.tests.tb3.MultiBrowserTest;
> 
> public class AbstractFieldElementSetValueReadOnlyTest extends MultiBrowserTest {
> ```

this only tests half of the components in the test UI

---

*[uitest/src/test/java/com/vaadin/tests/elements/abstracttextfield/CompatibilityAbstractFieldElementSetValueReadOnlyTest.java, line 15 at r3](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_cSIBTzYFgQi7qCABH:-K_cSIBTzYFgQi7qCABI:b-h5sxmv) ([raw file](https://github.com/vaadin/framework/blob/ad3a8851f58992099c897a905168db6829406f5a/uitest/src/test/java/com/vaadin/tests/elements/abstracttextfield/CompatibilityAbstractFieldElementSetValueReadOnlyTest.java#L15)):*
> ```Java
> import com.vaadin.tests.tb3.MultiBrowserTest;
> 
> public class CompatibilityAbstractFieldElementSetValueReadOnlyTest
> ```

this only tests half of the listed elements

---

*[uitest/src/test/java/com/vaadin/tests/elements/menubar/MenuBarUITest.java, line 39 at r3](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_cSyjzSnJLNbP8R2uR:-K_cSyk-BMMohdrzaqoL:b7ptfut) ([raw file](https://github.com/vaadin/framework/blob/ad3a8851f58992099c897a905168db6829406f5a/uitest/src/test/java/com/vaadin/tests/elements/menubar/MenuBarUITest.java#L39)):*
> ```Java
>     }
> 
>     // Tests against bug #14568
> ```

Should this be re-enabled as https://dev.vaadin.com/review/#/c/4618/ has been merged?

---

*[uitest/src/test/java/com/vaadin/tests/elements/notification/NotificationCloseTest.java, line 40 at r3](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_cfVw4cUGtTsqg1X3I:-K_cfVw4cUGtTsqg1X3J:b-5hl023) ([raw file](https://github.com/vaadin/framework/blob/ad3a8851f58992099c897a905168db6829406f5a/uitest/src/test/java/com/vaadin/tests/elements/notification/NotificationCloseTest.java#L40)):*
> ```Java
> 
>     private void testClose(int index) {
>         openTestURL("theme=reindeer");
> ```

why reindeer? probably not a blocker

---

*[uitest/src/test/java/com/vaadin/tests/elements/notification/NotificationGetTypeAndDescriptionTest.java, line 34 at r3](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_chkC2RhgPxdpMjwfH:-K_chkC2RhgPxdpMjwfI:b-adoz11) ([raw file](https://github.com/vaadin/framework/blob/ad3a8851f58992099c897a905168db6829406f5a/uitest/src/test/java/com/vaadin/tests/elements/notification/NotificationGetTypeAndDescriptionTest.java#L34)):*
> ```Java
>     // This method tests caption, description and type of the notification
>     private void testNotificationByIndex(int index) {
>         openTestURL("theme=reindeer");
> ```

(why reindeer?)

---

*[uitest/src/test/java/com/vaadin/tests/elements/table/GetRowHeaderTest.java, line 72 at r3](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_cf9QS_IB2_yjGdusx:-K_cf9QS_IB2_yjGdusy:bt48vwu) ([raw file](https://github.com/vaadin/framework/blob/ad3a8851f58992099c897a905168db6829406f5a/uitest/src/test/java/com/vaadin/tests/elements/table/GetRowHeaderTest.java#L72)):*
> ```Java
>          * Resize the handle back to initial value.
>          */
>         if (BrowserUtil.isIE(getDesiredCapabilities())) {
> ```

(this should probably be generalized as a column resize API in the element, but probably not in this patch)

---

*[uitest/src/test/java/com/vaadin/tests/elements/table/GetRowHeaderTest.java, line 73 at r3](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_cf75sxzMKobinuO2m:-K_cf75sxzMKobinuO2n:b-mu1uaj) ([raw file](https://github.com/vaadin/framework/blob/ad3a8851f58992099c897a905168db6829406f5a/uitest/src/test/java/com/vaadin/tests/elements/table/GetRowHeaderTest.java#L73)):*
> ```Java
>          */
>         if (BrowserUtil.isIE(getDesiredCapabilities())) {
>             int initHanldeWidth = handle.getSize().width;
> ```

(nit: handle)

---

*[uitest/src/test/java/com/vaadin/tests/elements/table/TableScrollTest.java, line 35 at r3](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_cejT51-F9h7C9ZDvb:-K_cejT64MVQtd36XsjI:b-vdw3l) ([raw file](https://github.com/vaadin/framework/blob/ad3a8851f58992099c897a905168db6829406f5a/uitest/src/test/java/com/vaadin/tests/elements/table/TableScrollTest.java#L35)):*
> ```Java
>         JavascriptExecutor js = getCommandExecutor();
>         String jsScript = "return arguments[0].getElementsByClassName(\"v-scrollable\")[0].scrollTop;";
>         if (BrowserUtil.isIE8(getDesiredCapabilities())) {
> ```

IE8 probably not relevant anymore

---

*[uitest/src/test/java/com/vaadin/tests/elements/table/TableScrollTest.java, line 45 at r3](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_ceotI-zJRbU6xWbN1:-K_ceotI-zJRbU6xWbN2:bb6wxig) ([raw file](https://github.com/vaadin/framework/blob/ad3a8851f58992099c897a905168db6829406f5a/uitest/src/test/java/com/vaadin/tests/elements/table/TableScrollTest.java#L45)):*
> ```Java
>         JavascriptExecutor js = getCommandExecutor();
>         String jsScript = "return arguments[0].getElementsByClassName(\"v-scrollable\")[0].scrollLeft;";
>         if (BrowserUtil.isIE8(getDesiredCapabilities())) {
> ```

IE8?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8117)*
<!-- Sent from Reviewable.io -->

Artur- commented:
Review status: 95 of 110 files reviewed at latest revision, 11 unresolved discussions.

---

*[uitest/src/main/java/com/vaadin/tests/elements/button/ButtonUI.java, line 34 at r3](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_cTNjI4CQ-BvafSrar:-K_cl_mdBKm14t4lbl42:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/ad3a8851f58992099c897a905168db6829406f5a/uitest/src/main/java/com/vaadin/tests/elements/button/ButtonUI.java#L34)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

I guess "quite" means "quiet" all over this file, but not a blocker
</blockquote></details>

Done.

---

*[uitest/src/test/java/com/vaadin/tests/elements/abstracttextfield/AbstractFieldElementSetValueReadOnlyTest.java, line 16 at r3](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_cRwVEFh9x6Mmteh4S:-K_dImAFsx-b2ezZRhow:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/ad3a8851f58992099c897a905168db6829406f5a/uitest/src/test/java/com/vaadin/tests/elements/abstracttextfield/AbstractFieldElementSetValueReadOnlyTest.java#L16)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

this only tests half of the components in the test UI
</blockquote></details>

Done.

---

*[uitest/src/test/java/com/vaadin/tests/elements/abstracttextfield/CompatibilityAbstractFieldElementSetValueReadOnlyTest.java, line 15 at r3](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_cSIBTzYFgQi7qCABH:-K_dHIwlVh6lk_Y-PRx3:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/ad3a8851f58992099c897a905168db6829406f5a/uitest/src/test/java/com/vaadin/tests/elements/abstracttextfield/CompatibilityAbstractFieldElementSetValueReadOnlyTest.java#L15)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

this only tests half of the listed elements
</blockquote></details>

Done.

---

*[uitest/src/test/java/com/vaadin/tests/elements/menubar/MenuBarUITest.java, line 39 at r3](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_cSyjzSnJLNbP8R2uR:-K_d5YfhikkNnPW-zpXs:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/ad3a8851f58992099c897a905168db6829406f5a/uitest/src/test/java/com/vaadin/tests/elements/menubar/MenuBarUITest.java#L39)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

Should this be re-enabled as https://dev.vaadin.com/review/#/c/4618/ has been merged?
</blockquote></details>

Done.

---

*[uitest/src/test/java/com/vaadin/tests/elements/notification/NotificationCloseTest.java, line 40 at r3](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_cfVw4cUGtTsqg1X3I:-K_dHRG2ZoQXX8BZVjOa:bkt9dnf) ([raw file](https://github.com/vaadin/framework/blob/ad3a8851f58992099c897a905168db6829406f5a/uitest/src/test/java/com/vaadin/tests/elements/notification/NotificationCloseTest.java#L40)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

why reindeer? probably not a blocker
</blockquote></details>

No idea, removed

---

*[uitest/src/test/java/com/vaadin/tests/elements/notification/NotificationGetTypeAndDescriptionTest.java, line 34 at r3](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_chkC2RhgPxdpMjwfH:-K_dHjk7C2e3o1q1TxUg:byr0zej) ([raw file](https://github.com/vaadin/framework/blob/ad3a8851f58992099c897a905168db6829406f5a/uitest/src/test/java/com/vaadin/tests/elements/notification/NotificationGetTypeAndDescriptionTest.java#L34)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

(why reindeer?)
</blockquote></details>

no idea, removed

---

*[uitest/src/test/java/com/vaadin/tests/elements/table/GetRowHeaderTest.java, line 72 at r3](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_cf9QS_IB2_yjGdusx:-K_dHt1qmnnIO_XEYGM4:bi5bw8e) ([raw file](https://github.com/vaadin/framework/blob/ad3a8851f58992099c897a905168db6829406f5a/uitest/src/test/java/com/vaadin/tests/elements/table/GetRowHeaderTest.java#L72)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

(this should probably be generalized as a column resize API in the element, but probably not in this patch)
</blockquote></details>

Could, probably not worth it for Table

---

*[uitest/src/test/java/com/vaadin/tests/elements/table/GetRowHeaderTest.java, line 73 at r3](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_cf75sxzMKobinuO2m:-K_dHqp2yF8Vw2GNNbye:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/ad3a8851f58992099c897a905168db6829406f5a/uitest/src/test/java/com/vaadin/tests/elements/table/GetRowHeaderTest.java#L73)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

(nit: handle)
</blockquote></details>

Done.

---

*[uitest/src/test/java/com/vaadin/tests/elements/table/TableScrollTest.java, line 35 at r3](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_cejT51-F9h7C9ZDvb:-K_dIjzdGriawnODHOl5:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/ad3a8851f58992099c897a905168db6829406f5a/uitest/src/test/java/com/vaadin/tests/elements/table/TableScrollTest.java#L35)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

IE8 probably not relevant anymore
</blockquote></details>

Done.

---

*[uitest/src/test/java/com/vaadin/tests/elements/table/TableScrollTest.java, line 45 at r3](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_ceotI-zJRbU6xWbN1:-K_dIkLHLtCyH2ff6lAG:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/ad3a8851f58992099c897a905168db6829406f5a/uitest/src/test/java/com/vaadin/tests/elements/table/TableScrollTest.java#L45)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

IE8?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8117)*
<!-- Sent from Reviewable.io -->

hesara commented:
Reviewed 11 of 109 files at r1, 1 of 9 files at r2, 1 of 2 files at r3, 14 of 14 files at r4.
Review status: all files reviewed at latest revision, 13 unresolved discussions.

---

*[testbench-api/src/main/java/com/vaadin/testbench/elements/ListSelectElement.java, line 70 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1We0PhgyvShL7YWvA:-Ka1We0PhgyvShL7YWvB:b-s2q951) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/testbench-api/src/main/java/com/vaadin/testbench/elements/ListSelectElement.java#L70)):*
> ```Java
>      *            the text of the option
>      */
>     public void deselectByText(String text) {
> ```

no readonly check?

---

*[testbench-api/src/main/java/com/vaadin/testbench/elements/TwinColSelectElement.java, line 66 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1XFvOJCBv3xBzRzTC:-Ka1XFvOJCBv3xBzRzTD:b-s2q951) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/testbench-api/src/main/java/com/vaadin/testbench/elements/TwinColSelectElement.java#L66)):*
> ```Java
> 
>     public void deselectByText(String text) {
>         selectedOptions.deselectAll();
> ```

no readonly check?

---

*[uitest/src/main/java/com/vaadin/tests/elements/TestBenchElementRightClick.java, line 13 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1XYRL-X8IKwn7S4NV:-Ka1XYRL-X8IKwn7S4NW:bdggks2) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/uitest/src/main/java/com/vaadin/tests/elements/TestBenchElementRightClick.java#L13)):*
> ```Java
> public class TestBenchElementRightClick extends AbstractTestUI {
> 
>     Table table = new Table();
> ```

(ideally would use something else than Table unless explicitly a Table test, but ok for now)

---

*[uitest/src/main/java/com/vaadin/tests/elements/abstracttextfield/AbstractTextElementSetValue.java, line 42 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1XwJ2adN93rU1tn3_:-Ka1XwJ2adN93rU1tn3a:b2jvy3n) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/uitest/src/main/java/com/vaadin/tests/elements/abstracttextfield/AbstractTextElementSetValue.java#L42)):*
> ```Java
>     int[] eventCounters = new int[comps.length + 1];
>     public static final String INITIAL_VALUE = "initial value";
>     public static final LocalDate INITIAL_DATE = LocalDate.now();
> ```

(not really ideal to use changing dates in tests)

---

*[uitest/src/main/java/com/vaadin/tests/elements/abstracttextfield/AbstractTextElementSetValue.java, line 64 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1Y6uQ0f1Q4wTBQfSD:-Ka1Y6uQ0f1Q4wTBQfSE:b-drcxge) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/uitest/src/main/java/com/vaadin/tests/elements/abstracttextfield/AbstractTextElementSetValue.java#L64)):*
> ```Java
>         addValueChangeListener(df, comps.length);
>         addComponent(df);
>         eventCountLabels[comps.length].setCaption("event  count");
> ```

(extra space in caption?)

---

*[uitest/src/main/java/com/vaadin/tests/elements/abstracttextfield/CompatibilityAbstractTextElementSetValue.java, line 45 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1aEhuIn0-8loTGx5v:-Ka1aEhvXbiuj-ZfFVId:by0fb1q) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/uitest/src/main/java/com/vaadin/tests/elements/abstracttextfield/CompatibilityAbstractTextElementSetValue.java#L45)):*
> ```Java
>     int[] eventCounters = new int[comps.length + 1];
>     public static final String INITIAL_VALUE = "initial value";
>     public static final Date INITIAL_DATE = Calendar.getInstance().getTime();
> ```

not really ideal to use a changing date in the test

---

*[uitest/src/test/java/com/vaadin/tests/elements/TestBenchElementClearValueTest.java, line 65 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1Y_KZKOI52PfzN00e:-Ka1Y_KZKOI52PfzN00f:b-1uaksk) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/uitest/src/test/java/com/vaadin/tests/elements/TestBenchElementClearValueTest.java#L65)):*
> ```Java
>     @Before
>     public void init() {
>         openTestURL("theme=reindeer");
> ```

why reindeer?

---

*[uitest/src/test/java/com/vaadin/tests/elements/abstracttextfield/CompatibilityAbstractFieldElementSetValueReadOnlyTest.java, line 72 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1afRPYuslrz4gux25:-Ka1afRPYuslrz4gux26:bi5q1ko) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/uitest/src/test/java/com/vaadin/tests/elements/abstracttextfield/CompatibilityAbstractFieldElementSetValueReadOnlyTest.java#L72)):*
> ```Java
>     }
> 
>     @Ignore("SliderElement.setValue does not exist")
> ```

(is there a ticket about this? see also the comment in the test which does this in the test class with JS)

---

*[uitest/src/test/java/com/vaadin/tests/elements/abstracttextfield/CompatibilityAbstractFieldElementSetValueReadOnlyTest.java, line 84 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1anDtFRQPDr5xokDT:-Ka1anDtFRQPDr5xokDU:b-kev3en) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/uitest/src/test/java/com/vaadin/tests/elements/abstracttextfield/CompatibilityAbstractFieldElementSetValueReadOnlyTest.java#L84)):*
> ```Java
>     }
> 
>     @Ignore("RichTextAreaElement does not have a setValue")
> ```

(is there a ticket about this?)

---

*[uitest/src/test/java/com/vaadin/tests/elements/abstracttextfield/CompatibilityAbstractTextElementSetValueTest.java, line 31 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1aPWOSirif-0vKFpf:-Ka1aPWOSirif-0vKFpg:b-5scqo) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/uitest/src/test/java/com/vaadin/tests/elements/abstracttextfield/CompatibilityAbstractTextElementSetValueTest.java#L31)):*
> ```Java
> 
> /**
>  *
> ```

(should remove javadoc or make it non-empty)

---

*[uitest/src/test/java/com/vaadin/tests/elements/slider/SliderGetHandleTest.java, line 14 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1ZbcST6fqu-95fdOK:-Ka1ZbcST6fqu-95fdOL:btwe8p9) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/uitest/src/test/java/com/vaadin/tests/elements/slider/SliderGetHandleTest.java#L14)):*
> ```Java
> 
>     @Test
>     public void testGetHanlde() {
> ```

Handle

---

*[uitest/src/test/java/com/vaadin/tests/elements/slider/SliderGetHandleTest.java, line 25 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1Zd_pqgMkMAfiTx6o:-Ka1Zd_pqgMkMAfiTx6p:bs02nsm) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/uitest/src/test/java/com/vaadin/tests/elements/slider/SliderGetHandleTest.java#L25)):*
> ```Java
>         // Handle has 0*0 size in Valo theme and requires some special treatment
>         ((JavascriptExecutor) driver).executeScript(
>                 "function simulate(f,c,d,e){var b,a=null;for(b in eventMatchers)if(eventMatchers[b].test(c)){a=b;break}if(!a)return!1;document.createEvent?(b=document.createEvent(a),a==\"HTMLEvents\"?b.initEvent(c,!0,!0):b.initMouseEvent(c,!0,!0,document.defaultView,0,d,e,d,e,!1,!1,!1,!1,0,null),f.dispatchEvent(b)):(a=document.createEventObject(),a.detail=0,a.screenX=d,a.screenY=e,a.clientX=d,a.clientY=e,a.ctrlKey=!1,a.altKey=!1,a.shiftKey=!1,a.metaKey=!1,a.button=1,f.fireEvent(\"on\"+c,a));return!0} var eventMatchers={HTMLEvents:/^(?:load|unload|abort|error|select|change|submit|reset|focus|blur|resize|scroll)$/,MouseEvents:/^(?:click|dblclick|mouse(?:down|up|over|move|out))$/}; "
> ```

this all is horrible, and should probably be in the element as valo is now default

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8117)*
<!-- Sent from Reviewable.io -->

Artur- commented:
Review status: all files reviewed at latest revision, 11 unresolved discussions.

---

*[testbench-api/src/main/java/com/vaadin/testbench/elements/ListSelectElement.java, line 70 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1We0PhgyvShL7YWvA:-Ka2HHzvqDm7UuzGMY-L:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/testbench-api/src/main/java/com/vaadin/testbench/elements/ListSelectElement.java#L70)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

no readonly check?
</blockquote></details>

Done.

---

*[testbench-api/src/main/java/com/vaadin/testbench/elements/TwinColSelectElement.java, line 66 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1XFvOJCBv3xBzRzTC:-Ka2GjrLdg-p9YcRuYrr:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/testbench-api/src/main/java/com/vaadin/testbench/elements/TwinColSelectElement.java#L66)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

no readonly check?
</blockquote></details>

Done.

---

*[uitest/src/main/java/com/vaadin/tests/elements/abstracttextfield/AbstractTextElementSetValue.java, line 42 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1XwJ2adN93rU1tn3_:-Ka2_7ftt31SuKpc75d2:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/uitest/src/main/java/com/vaadin/tests/elements/abstracttextfield/AbstractTextElementSetValue.java#L42)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

(not really ideal to use changing dates in tests)
</blockquote></details>

Done.

---

*[uitest/src/main/java/com/vaadin/tests/elements/abstracttextfield/AbstractTextElementSetValue.java, line 64 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1Y6uQ0f1Q4wTBQfSD:-Ka2_DfVZf7O244o2JEk:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/uitest/src/main/java/com/vaadin/tests/elements/abstracttextfield/AbstractTextElementSetValue.java#L64)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

(extra space in caption?)
</blockquote></details>

Done.

---

*[uitest/src/main/java/com/vaadin/tests/elements/abstracttextfield/CompatibilityAbstractTextElementSetValue.java, line 45 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1aEhuIn0-8loTGx5v:-Ka2aDjDByG9g-ue4ojB:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/uitest/src/main/java/com/vaadin/tests/elements/abstracttextfield/CompatibilityAbstractTextElementSetValue.java#L45)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

not really ideal to use a changing date in the test
</blockquote></details>

Done.

---

*[uitest/src/test/java/com/vaadin/tests/elements/TestBenchElementClearValueTest.java, line 65 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1Y_KZKOI52PfzN00e:-Ka2ZzY-3SLeLZTw4Bh-:b-4tcp00) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/uitest/src/test/java/com/vaadin/tests/elements/TestBenchElementClearValueTest.java#L65)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

why reindeer?
</blockquote></details>

Nobody knows

---

*[uitest/src/test/java/com/vaadin/tests/elements/abstracttextfield/CompatibilityAbstractFieldElementSetValueReadOnlyTest.java, line 72 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1afRPYuslrz4gux25:-Ka2GmxbO-pF0oEkwsxJ:b-xtt82i) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/uitest/src/test/java/com/vaadin/tests/elements/abstracttextfield/CompatibilityAbstractFieldElementSetValueReadOnlyTest.java#L72)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

(is there a ticket about this? see also the comment in the test which does this in the test class with JS)
</blockquote></details>

https://github.com/vaadin/framework/issues/8166

---

*[uitest/src/test/java/com/vaadin/tests/elements/abstracttextfield/CompatibilityAbstractFieldElementSetValueReadOnlyTest.java, line 84 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1anDtFRQPDr5xokDT:-Ka2GxxGaniFAJlKmO62:b-b3fgg0) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/uitest/src/test/java/com/vaadin/tests/elements/abstracttextfield/CompatibilityAbstractFieldElementSetValueReadOnlyTest.java#L84)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

(is there a ticket about this?)
</blockquote></details>

https://github.com/vaadin/testbench/issues/768, should be moved to the framework project

---

*[uitest/src/test/java/com/vaadin/tests/elements/abstracttextfield/CompatibilityAbstractTextElementSetValueTest.java, line 31 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1aPWOSirif-0vKFpf:-Ka2HCUMkr_em_d92w9o:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/uitest/src/test/java/com/vaadin/tests/elements/abstracttextfield/CompatibilityAbstractTextElementSetValueTest.java#L31)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

(should remove javadoc or make it non-empty)
</blockquote></details>

Done.

---

*[uitest/src/test/java/com/vaadin/tests/elements/slider/SliderGetHandleTest.java, line 14 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1ZbcST6fqu-95fdOK:-Ka2HGbsShFrTQKIWmay:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/uitest/src/test/java/com/vaadin/tests/elements/slider/SliderGetHandleTest.java#L14)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

Handle
</blockquote></details>

Done.

---

*[uitest/src/test/java/com/vaadin/tests/elements/slider/SliderGetHandleTest.java, line 25 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1Zd_pqgMkMAfiTx6o:-Ka1piniBppUraff8Nvl:b-fqe8aa) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/uitest/src/test/java/com/vaadin/tests/elements/slider/SliderGetHandleTest.java#L25)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

this all is horrible, and should probably be in the element as valo is now default
</blockquote></details>

Created https://github.com/vaadin/framework/issues/8166

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8117)*
<!-- Sent from Reviewable.io -->

hesara commented:
Reviewed 9 of 9 files at r5.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[testbench-api/src/main/java/com/vaadin/testbench/elements/TwinColSelectElement.java, line 68 at r5](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka69Yhh0sU9v-7tX-VK:-Ka69Yhh0sU9v-7tX-VL:b-84vkyu) ([raw file](https://github.com/vaadin/framework/blob/b9e504951c581e45245725410f334d2192a7e98b/testbench-api/src/main/java/com/vaadin/testbench/elements/TwinColSelectElement.java#L68)):*
> ```Java
>      * Removes the option with the given option text.
>      * <p>
>      * Any option already selected in the right column when this method is
> ```

is this accurate?

---

*[testbench-api/src/main/java/com/vaadin/testbench/elements/TwinColSelectElement.java, line 72 at r5](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka69W7uFg-1cNCONexV:-Ka69W7uFg-1cNCONexW:bot1miq) ([raw file](https://github.com/vaadin/framework/blob/b9e504951c581e45245725410f334d2192a7e98b/testbench-api/src/main/java/com/vaadin/testbench/elements/TwinColSelectElement.java#L72)):*
> ```Java
>      *
>      * @param text
>      *            the text of the option to select
> ```

to deselect?

---

*[uitest/src/test/java/com/vaadin/tests/elements/abstracttextfield/CompatibilityAbstractFieldElementSetValueReadOnlyTest.java, line 84 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1anDtFRQPDr5xokDT:-Ka6ATHWts_yRREmM43G:bdku3xv) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/uitest/src/test/java/com/vaadin/tests/elements/abstracttextfield/CompatibilityAbstractFieldElementSetValueReadOnlyTest.java#L84)):*
<details><summary><i>Previously, Artur- (Artur) wrote…</i></summary><blockquote>

https://github.com/vaadin/testbench/issues/768, should be moved to the framework project
</blockquote></details>

Copied to #8184, but I didn't have the permissions to close the original issue

---

*[uitest/src/test/java/com/vaadin/tests/elements/combobox/ComboBoxInputNotAllowedTest.java, line 16 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TQufFKTN_AimCLrjN:-Ka6B2xjdkRZdsMCRBce:b-48gu8x) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/test/java/com/vaadin/tests/elements/combobox/ComboBoxInputNotAllowedTest.java#L16)):*
<details><summary><i>Previously, Artur- (Artur) wrote…</i></summary><blockquote>

This is waiting for https://github.com/vaadin/testbench/pull/835, which needs to be fixed for all components and changed into a FW PR now
</blockquote></details>

A comment here about the issue would be nice - I'll see with Teemu on including 835 either in #8112 or separately.

---

*[uitest/src/test/java/com/vaadin/tests/elements/combobox/ComboBoxUITest.java, line 54 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TRbOESMWiKpOYjHWQ:-Ka6BDhw3FiRY43b4Fkn:b-34vmyd) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/test/java/com/vaadin/tests/elements/combobox/ComboBoxUITest.java#L54)):*
<details><summary><i>Previously, Artur- (Artur) wrote…</i></summary><blockquote>

same as above
</blockquote></details>

See above

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8117)*
<!-- Sent from Reviewable.io -->

tsuoanttila commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[testbench-api/src/main/java/com/vaadin/testbench/elements/CheckBoxGroupElement.java, line 62 at r3](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_ciSzb3T1rU-nadnd-:-Ka7MRpNkInNGMvz4lHx:b-sc0ezn) ([raw file](https://github.com/vaadin/framework/blob/0740ac116e637d81bd7e2415746f020dd25102e6/testbench-api/src/main/java/com/vaadin/testbench/elements/CheckBoxGroupElement.java#L62)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

(I hope no tests depend on this, probably not an issue)
</blockquote></details>

There was no tests in FW depending on this one at least. Some user might've used it, but not really that useful to begin with. I did exactly the same change in my PR..

---

*[testbench-api/src/main/java/com/vaadin/testbench/elements/TwinColSelectElement.java, line 68 at r5](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka69Yhh0sU9v-7tX-VK:-Ka7Mg1qRREsXJUodA8a:b-e552hu) ([raw file](https://github.com/vaadin/framework/blob/b9e504951c581e45245725410f334d2192a7e98b/testbench-api/src/main/java/com/vaadin/testbench/elements/TwinColSelectElement.java#L68)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

is this accurate?
</blockquote></details>

This JavaDoc could use some updating. Confusing at least if nothing else.

---

*[uitest/src/main/java/com/vaadin/tests/elements/TestBenchElementRightClick.java, line 13 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1XYRL-X8IKwn7S4NV:-Ka7MxqudWeJDJDD8HZQ:bsi4ek6) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/uitest/src/main/java/com/vaadin/tests/elements/TestBenchElementRightClick.java#L13)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

(ideally would use something else than Table unless explicitly a Table test, but ok for now)
</blockquote></details>

All components should contain context click events these days, so could test it with a layout for example. or the whole UI.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8117)*
<!-- Sent from Reviewable.io -->

tsuoanttila commented:
*[uitest/src/test/java/com/vaadin/tests/elements/combobox/ComboBoxInputNotAllowedTest.java, line 16 at r1](https://reviewable.io:443/reviews/vaadin/framework/8117#-K_TQufFKTN_AimCLrjN:-Ka7N_Pp31SA_GE3TI6B:bxs13s7) ([raw file](https://github.com/vaadin/framework/blob/b874bb77b8eddc67a7ef08f6ff112147089cf55c/uitest/src/test/java/com/vaadin/tests/elements/combobox/ComboBoxInputNotAllowedTest.java#L16)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

A comment here about the issue would be nice - I'll see with Teemu on including 835 either in #8112 or separately.
</blockquote></details>

I'd vote to do it with a separate PR just to avoid doing too much in either one of these. Enough changes as is.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8117)*
<!-- Sent from Reviewable.io -->

Artur- commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[testbench-api/src/main/java/com/vaadin/testbench/elements/TwinColSelectElement.java, line 68 at r5](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka69Yhh0sU9v-7tX-VK:-KaBo9lzjI6V_LrFT8pb:b8jrpyu) ([raw file](https://github.com/vaadin/framework/blob/b9e504951c581e45245725410f334d2192a7e98b/testbench-api/src/main/java/com/vaadin/testbench/elements/TwinColSelectElement.java#L68)):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

This JavaDoc could use some updating. Confusing at least if nothing else.
</blockquote></details>

Please suggest something better - maybe the problem is that "selected" here refers to a selection inside the right hand `<select>` and not items logically selected in the `TwinColSelect`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8117)*
<!-- Sent from Reviewable.io -->

Artur- commented:
Review status: 114 of 115 files reviewed at latest revision, 2 unresolved discussions.

---

*[testbench-api/src/main/java/com/vaadin/testbench/elements/TwinColSelectElement.java, line 68 at r5](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka69Yhh0sU9v-7tX-VK:-KaBow7kk0NTNSJcN7na:b-px5i3p) ([raw file](https://github.com/vaadin/framework/blob/b9e504951c581e45245725410f334d2192a7e98b/testbench-api/src/main/java/com/vaadin/testbench/elements/TwinColSelectElement.java#L68)):*
<details><summary><i>Previously, Artur- (Artur) wrote…</i></summary><blockquote>

Please suggest something better - maybe the problem is that "selected" here refers to a selection inside the right hand `<select>` and not items logically selected in the `TwinColSelect`
</blockquote></details>

Removed to avoid confusion

---

*[testbench-api/src/main/java/com/vaadin/testbench/elements/TwinColSelectElement.java, line 72 at r5](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka69W7uFg-1cNCONexV:-KaBpMHkB-6taxTnkOHC:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/b9e504951c581e45245725410f334d2192a7e98b/testbench-api/src/main/java/com/vaadin/testbench/elements/TwinColSelectElement.java#L72)):*
<details><summary><i>Previously, hesara (Henri Sara) wrote…</i></summary><blockquote>

to deselect?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8117)*
<!-- Sent from Reviewable.io -->

hesara commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> - any remaining changes to be done separately

---

Reviewed 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[uitest/src/main/java/com/vaadin/tests/elements/TestBenchElementRightClick.java, line 13 at r4](https://reviewable.io:443/reviews/vaadin/framework/8117#-Ka1XYRL-X8IKwn7S4NV:-KaCJiqf6IQYOOjUNwNR:bmv6lcu) ([raw file](https://github.com/vaadin/framework/blob/83c573f454d16118fb834f3cdde1c83c803c0444/uitest/src/main/java/com/vaadin/tests/elements/TestBenchElementRightClick.java#L13)):*
<details><summary><i>Previously, tsuoanttila (Teemu Suo-Anttila) wrote…</i></summary><blockquote>

All components should contain context click events these days, so could test it with a layout for example. or the whole UI.
</blockquote></details>

Can be addressed later, outside this PR

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8117#-:-KaCJzWhhQrtpN2JA-wh:b72coc9)*
<!-- Sent from Reviewable.io -->
